hadoop-gremlin/src/main/java/org/apache/tinkerpop/gremlin/hadoop/jsr223/HadoopGremlinPlugin.java

```diff
@@ -0,0 +1,153 @@
+import org.apache.tinkerpop.gremlin.hadoop.structure.util.ConfUtil;
+import org.apache.tinkerpop.gremlin.jsr223.AbstractGremlinPlugin;
+import org.apache.tinkerpop.gremlin.jsr223.BindingsCustomizer;
+import org.apache.tinkerpop.gremlin.jsr223.Customizer;
+import org.apache.tinkerpop.gremlin.jsr223.DefaultImportCustomizer;
+import org.apache.tinkerpop.gremlin.jsr223.ImportCustomizer;
+import org.apache.tinkerpop.gremlin.jsr223.LazyBindingsCustomizer;
+import org.apache.tinkerpop.gremlin.jsr223.console.ConsoleCustomizer;
+import org.apache.tinkerpop.gremlin.jsr223.console.GremlinShellEnvironment;
+import org.apache.tinkerpop.gremlin.jsr223.console.RemoteAcceptor;
+
+import javax.script.Bindings;
+import javax.script.SimpleBindings;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * @author Marko A. Rodriguez (http://markorodriguez.com)
+ */
+public final class HadoopGremlinPlugin extends AbstractGremlinPlugin {
+
+    protected static String NAME = "tinkerpop.hadoop";
+
+    private static final BindingsCustomizer bindings;
+
+    private static final ImportCustomizer imports;
+
+    private static final Set<String> appliesTo = new HashSet<>(Collections.singletonList("gremlin-groovy"));
+
+    static {
+        try {
+            // TODO: most of the imports here were wildcarded, but we dont' allow that anymore - needs review
+            imports = DefaultImportCustomizer.build()
+                    .addClassImports(
+                            Configuration.class,
+                            org.apache.hadoop.hdfs.DFSClient.class,
+                            FileSystem.class,
+                            ToolRunner.class,
+                            IOUtils.class,
+                            CodecPool.class,
+                            SequenceFileInputFormat.class,
+                            SequenceFileOutputFormat.class,
+                            Constants.class,
+                            HadoopConfiguration.class,
+                            HadoopEdge.class,
+                            HadoopElement.class,
+                            HadoopGraph.class,
+                            HadoopProperty.class,
+                            HadoopVertex.class,
+                            HadoopVertexProperty.class,
+                            ConfUtil.class,
+                            VertexWritable.class,
+                            GraphSONInputFormat.class,
+                            GraphSONOutputFormat.class,
+                            GraphSONRecordReader.class,
+                            GraphSONRecordWriter.class,
+                            GryoInputFormat.class,
+                            GryoOutputFormat.class,
+                            GryoRecordReader.class,
+                            GryoRecordWriter.class,
+                            ScriptInputFormat.class,
```

Comment:
We also need `ScriptOutputFormat.class`. And for safety, `ScriptRecordReader` and `ScriptRecordWriter`.